skin/frontend/base/default/js/bold/checkout_payment_booster/expresspay.js

```diff
@@ -711,6 +717,90 @@ const ExpressPay = async config => (async config => {
         }
     };
 
+    /**
+     * @param {Object} paymentData
+     * @returns void
+     */
+    const fixAddressEmailAddresses = paymentData => {
+        if (
+            !paymentData.billing_address.hasOwnProperty('emailAddress')
+            && paymentData.shipping_address.hasOwnProperty('emailAddress')
+        ) {
+            paymentData.billing_address.emailAddress = paymentData.shipping_address.emailAddress;
+        }
+
+        if (
+            !paymentData.billing_address.hasOwnProperty('emailAddress')
+            && paymentData.hasOwnProperty('customer')
+            && paymentData.customer.hasOwnProperty('email_address')
+        ) {
+            paymentData.billing_address.emailAddress = paymentData.customer.email_address;
```

Comment:
Could we set both the billing and shipping address email address in this second `if` and remove the 3rd one to clean things up? If the billing e-mail is still not set after the first `if` it follows that the shipping e-mail is also not set.

Reply:
I kept them separate to handle the edge case when only address or the other set, which likely won't happen, but should still be accounted for.

Reply:
Sounds good